shared/js/background/safari-events.es6.js

```diff
@@ -97,10 +93,42 @@ let handleMessage = (e) => {
     }
 }
 
+let getActiveTab = () => {
+    let activeTab = safari.application.activeBrowserWindow.activeTab
+    if (activeTab.ddgTabId) {
+        return tabManager.get({tabId: activeTab.ddgTabId})
+    } else {
+        let id = browserWrapper.getTabId({target: activeTab})
+        return tabManager.get({tabId: id})
+    }
+}
+
+let handleUIMessage = (req, res) => {
+    if (req.getCurrentTab || req.getTab) {
+        res(getActiveTab())
+    } else if (req.getTopBlocked) {
+        res(Companies.getTopBlocked(req.getTopBlocked))
+    } else if (req.getBrowser) {
+        res('safari')
+    } else if (req.whitelisted) {
+        res(tabManager.whitelistDomain(req.whitelisted))
+    } else if (req.getSiteScore) {
+        let tab = tabManager.get({tabId: req.getSiteScore})
+        if (tab) res(tab.site.score.get())
+    } else if (req.getTopBlockedByPages) {
+        res(Companies.getTopBlockedByPages(req.getTopBlockedByPages))
+    } else if (req.resetTrackersData) {
+        Companies.resetData()
+        safari.self.hide()
+    }
+}
```

Comment:
The idea here is, move the stuff that belongs in the background process to the background. That way you wouldn't need to expose `tabManager` and `companies` in any way.